### ISSUE TITLE
have to repeatedly invoke, different "failed to download wheel" each time

### ISSUE THREAD

**Issue by smemsh** [ddelange/pipgrip#55]
#### What you were trying to do (and why)

determine package dependencies using a test case
```
pipgrip --tree pipgrip
```
#### What happened (including command output)

each time it was invoked, it gave a traceback about "failed to download wheel" with one of the dependencies, but then the next time it would proceed to the next dependency.  after all dependencies were downloaded, it worked.  so for example with pipgrip, it would say in turn:

```
RuntimeError: Failed to download wheel for pkginfo>=1.4.2
RuntimeError: Failed to download wheel for packaging>=17
RuntimeError: Failed to download wheel for click>=7
RuntimeError: Failed to download wheel for anytree
RuntimeError: Failed to download wheel for pyparsing>=2.0.2
RuntimeError: Failed to download wheel for six>=1.9.0
```
then it finally succeeded:
```
 $ pipgrip --tree pipgrip
pipgrip (0.6.8)
├── anytree (2.8.0)
│   └── six>=1.9.0 (1.16.0)
├── click>=7 (8.0.0)
├── packaging>=17 (20.9)
│   └── pyparsing>=2.0.2 (2.4.7)
├── pip>=7.1.0 (21.1.1)
├── pkginfo>=1.4.2 (1.7.0)
├── setuptools>=38.3 (56.2.0)
└── wheel (0.36.2)
```

(there might have been one or more that was missed above because it had scrolled off my screen, I think it was every dependency)

I did manage to capture the one with `-vvv` for setuptools:

```
 $ pipgrip -vvv --tree pipgrip
DEBUG: Environment: {'implementation_name': 'cpython', 'implementation_version': '3.9.5', 'os_name': 'posix', 'platform_machine': 'x86_64', 'platform_release': '5.6.0-1055-oem', 'platform_system': 'Linux', 'platform_version': '#59-Ubuntu SMP Fri Apr 16 08:43:28 UTC 2021', 'python_full_version': '3.9.5', 'platform_python_implementation': 'CPython', 'python_version': '3.9', 'sys_platform': 'linux'}
DEBUG: Pip version: [21, 1, 1]
DEBUG: Downloading/building wheel for pipgrip
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--wheel-dir', '/home/scott/.cache/pip/wheels/pipgrip', '--progress-bar=off', 'pipgrip']
DEBUG: {'pipgrip': '/home/scott/.cache/pip/wheels/pipgrip/pipgrip-0.6.8-py2.py3-none-any.whl'}
DEBUG: Searching metadata in /home/scott/.cache/pip/wheels/pipgrip/pipgrip-0.6.8-py2.py3-none-any.whl
DEBUG: dropped conditional dep enum34 ; python_version == "2.7"
DEBUG: dropped conditional dep typing ; python_version == "2.7"
DEBUG: Finding possible versions for pipgrip
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--progress-bar=off', 'pipgrip==42.42.post424242']
DEBUG: {'pipgrip': ['0.0.1', '0.0.2', '0.0.3', '0.1.0', '0.2.0', '0.2.1', '0.2.2', '0.3.0', '0.3.1', '0.4.0', '0.4.1', '0.4.2', '0.5.0', '0.5.1', '0.6.0', '0.6.1', '0.6.2', '0.6.3', '0.6.4', '0.6.5', '0.6.6', '0.6.7', '0.6.8']}
INFO: fact: _root_ is root
INFO: derived: root
INFO: fact: root depends on pipgrip (*)
INFO: selecting . (0.0.0)
INFO: derived: pipgrip (*)
INFO: fact: pipgrip (0.6.8) depends on anytree (*)
INFO: fact: pipgrip (0.6.8) depends on click (>=7)
INFO: fact: pipgrip (0.6.8) depends on packaging (>=17)
INFO: fact: pipgrip (0.6.8) depends on pip (>=7.1.0)
INFO: fact: pipgrip (0.6.8) depends on pkginfo (>=1.4.2)
INFO: fact: pipgrip (0.6.8) depends on setuptools (>=38.3)
INFO: fact: pipgrip (0.6.8) depends on wheel (*)
INFO: selecting pipgrip (0.6.8)
INFO: derived: wheel (*)
INFO: derived: setuptools (>=38.3)
INFO: derived: pkginfo (>=1.4.2)
INFO: derived: pip (>=7.1.0)
INFO: derived: packaging (>=17)
INFO: derived: click (>=7)
INFO: derived: anytree (*)
DEBUG: Downloading/building wheel for wheel
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--wheel-dir', '/home/scott/.cache/pip/wheels/pipgrip', '--progress-bar=off', 'wheel']
DEBUG: {'wheel': '/home/scott/.cache/pip/wheels/pipgrip/wheel-0.36.2-py2.py3-none-any.whl'}
DEBUG: Searching metadata in /home/scott/.cache/pip/wheels/pipgrip/wheel-0.36.2-py2.py3-none-any.whl
DEBUG: dropped conditional dep pytest (>=3.0.0) ; extra == 'test'
DEBUG: dropped conditional dep pytest-cov ; extra == 'test'
DEBUG: Finding possible versions for wheel
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--progress-bar=off', 'wheel==42.42.post424242']
DEBUG: {'wheel': ['0.1', '0.2', '0.3', '0.4', '0.4.1', '0.4.2', '0.5', '0.6', '0.7', '0.8', '0.9', '0.9.1', '0.9.2', '0.9.3', '0.9.4', '0.9.5', '0.9.6', '0.9.7', '0.10.0', '0.10.1', '0.10.2', '0.10.3', '0.11.0', '0.12.0', '0.13.0', '0.14.0', '0.15.0', '0.16.0', '0.17.0', '0.18.0', '0.19.0', '0.21.0', '0.22.0', '0.23.0', '0.24.0', '0.25.0', '0.26.0', '0.27.0', '0.28.0', '0.29.0', '0.30.0', '0.31.0', '0.31.1', '0.32.0', '0.32.1', '0.32.2', '0.32.3', '0.33.0', '0.33.1', '0.33.4', '0.33.5', '0.33.6', '0.34.0', '0.34.1', '0.34.2', '0.35.0', '0.35.1', '0.36.0', '0.36.1', '0.36.2']}
DEBUG: Downloading/building wheel for setuptools>=38.3
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--wheel-dir', '/home/scott/.cache/pip/wheels/pipgrip', '--progress-bar=off', 'setuptools>=38.3']
DEBUG: {'setuptools>=38.3': '/home/scott/.cache/pip/wheels/pipgrip/setuptools-56.2.0-py3-none-any.whl'}
DEBUG: Searching metadata in /home/scott/.cache/pip/wheels/pipgrip/setuptools-56.2.0-py3-none-any.whl
DEBUG: dropped conditional dep certifi (==2016.9.26) ; extra == 'certs'
DEBUG: dropped conditional dep sphinx ; extra == 'docs'
DEBUG: dropped conditional dep jaraco.packaging (>=8.2) ; extra == 'docs'
DEBUG: dropped conditional dep rst.linker (>=1.9) ; extra == 'docs'
DEBUG: dropped conditional dep pygments-github-lexers (==0.0.5) ; extra == 'docs'
DEBUG: dropped conditional dep sphinx-inline-tabs ; extra == 'docs'
DEBUG: dropped conditional dep wincertstore (==0.2) ; (sys_platform == "win32") and extra == 'ssl'
DEBUG: dropped conditional dep pytest (>=4.6) ; extra == 'testing'
DEBUG: dropped conditional dep pytest-checkdocs (>=2.4) ; extra == 'testing'
DEBUG: dropped conditional dep pytest-flake8 ; extra == 'testing'
DEBUG: dropped conditional dep pytest-cov ; extra == 'testing'
DEBUG: dropped conditional dep pytest-enabler (>=1.0.1) ; extra == 'testing'
DEBUG: dropped conditional dep mock ; extra == 'testing'
DEBUG: dropped conditional dep flake8-2020 ; extra == 'testing'
DEBUG: dropped conditional dep virtualenv (>=13.0.0) ; extra == 'testing'
DEBUG: dropped conditional dep pytest-virtualenv (>=1.2.7) ; extra == 'testing'
DEBUG: dropped conditional dep wheel ; extra == 'testing'
DEBUG: dropped conditional dep paver ; extra == 'testing'
DEBUG: dropped conditional dep pip (>=19.1) ; extra == 'testing'
DEBUG: dropped conditional dep jaraco.envs ; extra == 'testing'
DEBUG: dropped conditional dep pytest-xdist ; extra == 'testing'
DEBUG: dropped conditional dep sphinx ; extra == 'testing'
DEBUG: dropped conditional dep jaraco.path (>=3.2.0) ; extra == 'testing'
DEBUG: dropped conditional dep pytest-black (>=0.3.7) ; (platform_python_implementation != "PyPy" and python_version < "3.10") and extra == 'testing'
DEBUG: dropped conditional dep pytest-mypy ; (platform_python_implementation != "PyPy" and python_version < "3.10") and extra == 'testing'
DEBUG: Finding possible versions for setuptools
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--progress-bar=off', 'setuptools==42.42.post424242']
DEBUG: {'setuptools': ['0.7.2', '0.7.3', '0.7.4', '0.7.5', '0.7.6', '0.7.7', '0.7.8', '0.8', '0.9', '0.9.1', '0.9.2', '0.9.3', '0.9.4', '0.9.5', '0.9.6', '0.9.7', '0.9.8', '1.0', '1.1', '1.1.1', '1.1.2', '1.1.3', '1.1.4', '1.1.5', '1.1.6', '1.1.7', '1.2', '1.3', '1.3.1', '1.3.2', '1.4', '1.4.1', '1.4.2', '2.0', '2.0.1', '2.0.2', '2.1', '2.1.1', '2.1.2', '2.2', '3.0', '3.0.1', '3.0.2', '3.1', '3.2', '3.3', '3.4', '3.4.1', '3.4.2', '3.4.3', '3.4.4', '3.5', '3.5.1', '3.5.2', '3.6', '3.7', '3.7.1', '3.8', '3.8.1', '4.0', '4.0.1', '5.0', '5.0.1', '5.0.2', '5.1', '5.2', '5.3', '5.4', '5.4.1', '5.4.2', '5.5', '5.5.1', '5.6', '5.7', '5.8', '6.0.1', '6.0.2', '6.1', '7.0', '8.0', '8.0.1', '8.0.2', '8.0.3', '8.0.4', '8.1', '8.2', '8.2.1', '8.3', '9.0', '9.0.1', '9.1', '10.0', '10.0.1', '10.1', '10.2', '10.2.1', '11.0', '11.1', '11.2', '11.3', '11.3.1', '12.0', '12.0.1', '12.0.2', '12.0.3', '12.0.4', '12.0.5', '12.1', '12.2', '12.3', '12.4', '13.0.1', '13.0.2', '14.0', '14.1', '14.1.1', '14.2', '14.3', '14.3.1', '15.0', '15.1', '15.2', '16.0', '17.0', '17.1', '17.1.1', '18.0', '18.0.1', '18.1', '18.2', '18.3', '18.3.1', '18.3.2', '18.4', '18.5', '18.6', '18.6.1', '18.7', '18.7.1', '18.8', '18.8.1', '19.0', '19.1', '19.1.1', '19.2', '19.3', '19.4', '19.4.1', '19.5', '19.6', '19.6.1', '19.6.2', '19.7', '20.0', '20.1', '20.1.1', '20.2.2', '20.3', '20.3.1', '20.4', '20.6.6', '20.6.7', '20.6.8', '20.7.0', '20.8.0', '20.8.1', '20.9.0', '20.10.1', '21.0.0', '21.1.0', '21.2.0', '21.2.1', '21.2.2', '22.0.0', '22.0.1', '22.0.2', '22.0.4', '22.0.5', '23.0.0', '23.1.0', '23.2.0', '23.2.1', '24.0.0', '24.0.1', '24.0.2', '24.0.3', '24.1.0', '24.1.1', '24.2.0', '24.2.1', '24.3.0', '24.3.1', '25.0.0', '25.0.1', '25.0.2', '25.1.0', '25.1.1', '25.1.2', '25.1.3', '25.1.4', '25.1.5', '25.1.6', '25.2.0', '25.3.0', '25.4.0', '26.0.0', '26.1.0', '26.1.1', '27.0.0', '27.1.0', '27.1.2', '27.2.0', '27.3.0', '27.3.1', '28.0.0', '28.1.0', '28.2.0', '28.3.0', '28.4.0', '28.5.0', '28.6.0', '28.6.1', '28.7.0', '28.7.1', '28.8.0', '28.8.1', '29.0.0', '29.0.1', '30.0.0', '30.1.0', '30.2.0', '30.2.1', '30.3.0', '30.4.0', '31.0.0', '31.0.1', '32.0.0', '32.1.0', '32.1.1', '32.1.2', '32.1.3', '32.2.0', '32.3.0', '32.3.1', '33.1.0', '33.1.1', '34.0.0', '34.0.1', '34.0.2', '34.0.3', '34.1.0', '34.1.1', '34.2.0', '34.3.0', '34.3.1', '34.3.2', '34.3.3', '34.4.0', '34.4.1', '35.0.0', '35.0.1', '35.0.2', '36.0.1', '36.1.0', '36.1.1', '36.2.0', '36.2.1', '36.2.2', '36.2.3', '36.2.4', '36.2.5', '36.2.6', '36.2.7', '36.3.0', '36.4.0', '36.5.0', '36.6.0', '36.6.1', '36.7.0', '36.7.1', '36.7.2', '36.8.0', '37.0.0', '38.0.0', '38.1.0', '38.2.0', '38.2.1', '38.2.3', '38.2.4', '38.2.5', '38.3.0', '38.4.0', '38.4.1', '38.5.0', '38.5.1', '38.5.2', '38.6.0', '38.6.1', '38.7.0', '39.0.0', '39.0.1', '39.1.0', '39.2.0', '40.0.0', '40.1.0', '40.1.1', '40.2.0', '40.3.0', '40.4.0', '40.4.1', '40.4.2', '40.4.3', '40.5.0', '40.6.0', '40.6.1', '40.6.2', '40.6.3', '40.7.0', '40.7.1', '40.7.2', '40.7.3', '40.8.0', '40.9.0', '41.0.0', '41.0.1', '41.1.0', '41.2.0', '41.3.0', '41.4.0', '41.5.0', '41.5.1', '41.6.0', '42.0.0', '42.0.1', '42.0.2', '43.0.0', '44.0.0', '44.1.0', '44.1.1', '45.0.0', '45.1.0', '45.2.0', '45.3.0', '46.0.0', '46.1.0', '46.1.1', '46.1.2', '46.1.3', '46.2.0', '46.3.0', '46.3.1', '46.4.0', '47.0.0', '47.1.0', '47.1.1', '47.2.0', '47.3.0', '47.3.1', '47.3.2', '48.0.0', '49.0.0', '49.0.1', '49.1.0', '49.1.1', '49.1.2', '49.1.3', '49.2.0', '49.2.1', '49.3.0', '49.3.1', '49.3.2', '49.4.0', '49.5.0', '49.6.0', '50.0.0', '50.0.1', '50.0.2', '50.0.3', '50.1.0', '50.2.0', '50.3.0', '50.3.1', '50.3.2', '51.0.0', '51.1.0', '51.1.1', '51.1.2', '51.2.0', '51.3.0', '51.3.1', '51.3.2', '51.3.3', '52.0.0', '53.0.0', '53.1.0', '54.0.0', '54.1.0', '54.1.1', '54.1.2', '54.1.3', '54.2.0', '56.0.0', '56.1.0', '56.2.0']}
DEBUG: Downloading/building wheel for pkginfo>=1.4.2
DEBUG: ['/opt/python-3.9.5/bin/python3.9', '-m', 'pip', 'wheel', '--no-deps', '--disable-pip-version-check', '--wheel-dir', '/home/scott/.cache/pip/wheels/pipgrip', '--progress-bar=off', 'pkginfo>=1.4.2']
ERROR: Failed to extract wheel filename from pip stdout:
Collecting pkginfo>=1.4.2
  Using cached pkginfo-1.7.0-py2.py3-none-any.whl (25 kB)
Saved ./.cache/pip/wheels/pipgrip/pkginfo-1.7.0-py2.py3-none-any.whl
Traceback (most recent call last):
  File "/home/scott/.local/bin/pipgrip", line 8, in <module>
    sys.exit(main())
  File "/home/scott/.local/lib/python3.9/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/home/scott/.local/lib/python3.9/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/home/scott/.local/lib/python3.9/site-packages/click/core.py", line 1401, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/scott/.local/lib/python3.9/site-packages/click/core.py", line 767, in invoke
    return __callback(*args, **kwargs)
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/cli.py", line 426, in main
    solution = solver.solve()
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/version_solver.py", line 69, in solve
    if not self._run():
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/version_solver.py", line 85, in _run
    next_package = self._choose_package_version()
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/version_solver.py", line 349, in _choose_package_version
    term = self._next_term_to_try()
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/version_solver.py", line 337, in _next_term_to_try
    term = min(*unsatisfied, key=_get_min)
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/version_solver.py", line 324, in _get_min
    versions = self._source.versions_for(
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/libs/mixology/package_source.py", line 77, in versions_for
    return self._versions_for(package, constraint)
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/package_source.py", line 167, in _versions_for
    self.discover_and_add(package.req.__str__())
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/package_source.py", line 129, in discover_and_add
    to_create = discover_dependencies_and_versions(
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/pipper.py", line 395, in discover_dependencies_and_versions
    wheel_fname = _download_wheel(
  File "/home/scott/.local/lib/python3.9/site-packages/pipgrip/pipper.py", line 331, in _download_wheel
    raise RuntimeError("Failed to download wheel for {}".format(package))
RuntimeError: Failed to download wheel for pkginfo>=1.4.2
```
this is just ubuntu 20 with a compiled python-3.9.5, and --user install of pipgrip-0.6.8.  there would have been no other system libraries installed, only the ones that came with python:

```
 $ pip3 freeze --all 
anytree==2.8.0
click==8.0.0
packaging==20.9
pip==21.1.1
pipgrip==0.6.8
pkginfo==1.7.0
pyparsing==2.4.7
setuptools==56.0.0
six==1.16.0
wheel==0.36.2
```

**Comment by ddelange** [comment]
Hi @smemsh, 

Can you try with `pip install git+https://github.com/ddelange/pipgrip@relative-cwd-wheel-output`?

If that works for you, I'll merge and release

**Comment by smemsh** [comment]
seems to work, tried with a few random packages that were never seen before, worked fine.  thanks.  (boy does tensorflow take a while ;-)

**Comment by ddelange** [comment]
Released [`0.6.9`](https://github.com/ddelange/pipgrip/releases/tag/0.6.9) 

Thanks for the bug report!
